api/src/routes/donate.test.ts

```diff
@@ -133,6 +173,17 @@ describe('Donate', () => {
           error: 'Donation failed due to a server error.'
         });
       });
+
+      it('should return 400 if user has not completed challenges', async () => {
+        await fastifyTestInstance.prisma.user.updateMany({
+          where: { email: userWithProgress.email },
+          data: userWithoutProgress
+        });
+        const failResponse = await superPost('/donate/charge-stripe-card').send(
+          chargeStripeCardReqBody
+        );
+        expect(failResponse.status).toBe(400);
```

Comment:
Could you add the expected response body, too? It's nice to assert the status code, but the thing we really care about is the body.

Reply:
Thanks for the feedback. I included the body in the test accrodingly.